### PR TITLE
Add ability to pass a role assumed by Cloud Formation.

### PR DESCRIPTION
**Pull request by Nikokrock**
This mainly allows to separate rights for users and rights for Cloud
Formation itself.

### CODE REVIEW

File: src/e3/aws/__init__.py [hunk 1]
```diff
@@ -428,14 +428,24 @@ def resources(self) -> List[AWSObject]:
 class Stack(cfn.Stack, Construct):
     """Cloudformation stack using troposphere resources."""
 
-    def __init__(self, stack_name: str, description: Optional[str] = None) -> None:
+    def __init__(
+        self,
+        stack_name: str,
+        cfn_role_arn: Optional[str],
```
Comment (enzbang) [inline src/e3/aws/__init__.py:434]: Set to None by default?

File: src/e3/aws/__init__.py [hunk 2]
```diff
@@ -428,14 +428,24 @@ def resources(self) -> List[AWSObject]:
...
+        description: Optional[str] = None,
+    ) -> None:
         """Initialize Stack attributes.
 
         :param stack_name: stack name
         :param description: a description of the stack
+        :param cfn_role_arn: role asssumed by cloud formation to create the stack
```
Comment (enzbang) [inline src/e3/aws/__init__.py:441]: can you move that line above :param description?

File: src/e3/aws/cfn/__init__.py [hunk 3]
```diff
@@ -265,13 +266,17 @@ def export(self):
...
-    def __init__(self, name, description=None):
+    def __init__(self, name, cfn_role_arn=None, description=None):
         """Initialize a stack.
 
         :param name: stack name
         :type name: str
         :param description: a description of the stack
         :type description: str | None
+        :param cfn_role_arn: Arn of the role to be assumed by cloudformation. If
+            None then use user role. In the future role_arn should be mandatary in
```
Comment (enzbang) [inline src/e3/aws/cfn/__init__.py:277]: mandatory

**Review by enzbang — CHANGES_REQUESTED** [review]

**Review by enzbang — DISMISSED** [review]

File: src/e3/aws/cfn/__init__.py [hunk 4]
```diff
@@ -2,6 +2,7 @@
 from enum import Enum
 import re
 import yaml
+import logging
```
Comment (pierretr) [inline src/e3/aws/cfn/__init__.py:5]: logging does not seem to be used.

**Review by pierretr — APPROVED** [review]
LGTM.

**Pull request merged**